### DISCUSSION TITLE
A discussion about the "realism feature" and throwin my hat in the ring.

### DISCUSSION THREAD

**Discussion by zad112** [CosmicHikari/Weather2-Remastered#27] [General]
Mainly just wanted to suggest some ways the large scale dynamics of the storms motion and development can play out. The easiest way I can think would be to have different biomes have different average temperatures and different levels of humidity. use the list from here they have it all layed out. Having day time to night time variations is neat but not really needed. Different air masses bumping into each other create different weather. heres a little chart with all the weather combos, also a little note atlest on earth cold is king so warm air must move out of cold airs way. 
However if there is a global wind the cold air can be pushed away allowing warm air to fill the void.

-winds cold --> warm biome FAST (cold king)
cold and dry --> warm and dry = high winds no precipitation some high altitude clouds
cold and wet --> warm and dry = Fog, low winds, no precipitation. low altitude clouds, small storms possible the next day.
cold and dry --> warm and wet = high winds, high precipitation (cold front), high chance for tornadoes, thick clouds. fast moving.
cold and wet --> warm and wet = mid winds, mid precipitation, slow moving, no severe weather, thick clouds. if cold SNOW.

-winds warm --> cold Slow (warm moves when the cold moves)
Warm and dry --> cold and dry = dust storms high winds, no precipitation high altitude clouds.
warm and wet --> cold and dry = (warm front) slow moving, starts as high altitude clouds and descends until its fog then passes. mild precipitation,
Warm and dry --> cold and wet = increasing temperature, (dry line) isolated small storms, sever weather, very powerful tornadoes possible.
warm and wet --> cold and wet = (don't have too munch info about this one as its rare on land) light rain, thick stratus clouds. light winds very slow moving.

airmases can also be modified by the terrain they are over. as a cold dry airmass moves over a mild mid humidity plain it can pick up some of the temperature and moisture making it and cool moist airmass.

now for the fun part how to turn real life into mine craft coolness. I believe most of this can be done using the storm system already implemented by allowing the systems to interact with the biomes they are over. A storm can be assumed to be a "moving airmass" and each biome can have a natural inherent "airmass". Also this inherent airmas can become a storm and move around based on a random tick. When a storm that has the property of a cold dry airmass is blown by global winds over a biome that is warm and humid you can has it increase massively in intensity, and if a warm tropical cyclone moves over a cold tundra biome it will lose strength quickly. The only major change really that would need to happen is A a much larger scale as airmases are the size of biomes and/or B allowing storm size to scale to world size (amplified biomes) making much much larger storms. 

thats all I have for now atlest in suggestions n stuff. love how its coming along tho.

ALSO if you need any help with textures(any kind) and basic models I can help just send me whatever and ill show you what I can do. I love this project and wana bring to to the spot light.

**Comment by CosmicHikari** [comment]
Just saw this - wanted to post to let you know it hasn't been ignored or condemned to "not possible" land lol.

Temperature, humidity, wind speed, and fronts are being added in a future update, as well as many improvements to the particles, etc. I am sure @Mrbt0907 would like to review this to see if there's anything you've mentioned that can help contribute to what he's already implemented.

edit: typos lol

**Comment by Mrbt0907** [comment]
Oh wow, I honestly did not see this post, I am sorry.

But yes, I have something similar to this in the game right now. With the new front system, I can emulate this behavior pretty well now. I didn't code in the specific behaviors just yet, but I thank you for laying out a perfect way to do air masses. This will help speed up the update.
